### PR TITLE
task solution

### PR DESCRIPTION
**Pull request by HubertKlamann**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,29 @@
...
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```
Comment (Matylda-Lewandowska) [inline src/transformStateWithClones.js:28]: I think after the cases, there should be `default:` with an error just like in the previous task, but I'm not sure. ;)

**Review by Matylda-Lewandowska — APPROVED** [review]
This code looks fine ;)

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  let newState = JSON.parse(JSON.stringify(state));
```
Comment (asakevych) [inline src/transformStateWithClones.js:11]: please read checklist point 1 - use Object.assign or spread operator

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,29 @@
...
+
+  for (const obj of actions) {
```
Comment (asakevych) [inline src/transformStateWithClones.js:13]: please give your variables discriptive names -> ` action of actions` would be better

File: src/transformStateWithClones.js (hunk 1 above)
Comment (asakevych) [reply src/transformStateWithClones.js:28]: yes, @Matylda-Lewandowska is right, default is required 😉

File: src/transformStateWithClones.js [hunk 4]
```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
```
Comment (asakevych) [inline src/transformStateWithClones.js:10]: please use descriptive names - try `statesHistory` instead of `arr`

**Review by asakevych — CHANGES_REQUESTED** [review]
Good job, but some changes are required 😉

**Review by choeqq — CHANGES_REQUESTED** [review]
Waiting for any updates @HubertKlamann 👀

**Review by choeqq — APPROVED** [review]
Nicely done ✅

**Pull request closed without merging**